production/mimir/mimir.libsonnet

```diff
@@ -0,0 +1,46 @@
+(import 'ksonnet-util/kausal.libsonnet') +
+(import 'configmap.libsonnet') +
+(import 'config.libsonnet') +
+{
+  local container = $.core.v1.container,
+  local containerPort = $.core.v1.containerPort,
+  local volumeMount = $.core.v1.volumeMount,
+  local deployment = $.apps.v1.deployment,
+  local volume = $.core.v1.volume,
+  local service = $.core.v1.service,
+  local servicePort = service.mixin.spec.portsType,
+
+  local mimir_config_volume = 'mimir-conf',
+
+  namespace: $.core.v1.namespace.new($._config.namespace),
+
+  mimir_container::
+    container.new('mimir', $._images.mimir) +
+    container.withPorts([
+      containerPort.new('prom-metrics', $._config.http.port),
+    ]) +
+    container.withArgs([
+      '-config.file=/conf/mimir.yaml',
+    ]) +
+    container.withVolumeMounts([
+      volumeMount.new(mimir_config_volume, '/conf'),
+    ]),
```

Comment:
You probably also want to create and configure a volume for the filesystem storage, Otherwise I'm not sure if the data will persist across restarts etc. (https://grafana.com/docs/mimir/latest/operators-guide/configure/reference-configuration-parameters/#filesystem_storage_backend).

Reply:
Good catch. I'll add that in.

Reply:
I've added something that seems to work in 09f9254 - is there anything I've missed, particularly in the config for Mimir? (Not aiming to create a robust production-ready setup here of course)